### PR TITLE
Feature/various

### PR DESCRIPTION
**Pull request by martinfalk**
Some minor fixes
* fixed content type for the image save dialog, making it remember the last folder
* minor optimization in the raycasting in connection with rendering on top of the background
* fixes for the color map legend (initial layouting and new background mode "solid color")

image

### CODE REVIEW

File: modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h [hunk 1]
```diff
@@ -65,11 +65,14 @@ class IVW_MODULE_BRUSHINGANDLINKING_API BrushingAndLinkingManager {
     bool isColumnSelected(size_t column) const;
 
     void setSelected(const BrushingAndLinkingInport* src, const std::unordered_set<size_t>& idx);
+    void clearSelected();
```
Comment (petersteneteg) [inline modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: what's the thinking here? why not use a empty set together with setSelected? How will this keep track of the sources?

File: modules/brushingandlinking/src/brushingandlinkingmanager.cpp [hunk 2]
```diff
@@ -66,17 +66,29 @@ void BrushingAndLinkingManager::setSelected(const BrushingAndLinkingInport*,
...
+    selected_.clear();
+    owner_->invalidate(invalidationLevel_);
+}
+
 void BrushingAndLinkingManager::setFiltered(const BrushingAndLinkingInport* src,
                                             const std::unordered_set<size_t>& indices) {
     filtered_.set(src, indices);
 }
 
+void BrushingAndLinkingManager::clearFiltered() { filtered_.clear(); }
```
Comment (petersteneteg) [inline modules/brushingandlinking/src/brushingandlinkingmanager.cpp:79]: invalidation?

File: modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h (hunk 1 above)
Comment (petersteneteg) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: should this be public? a processor using the manager should not call this right?

**Review by petersteneteg — APPROVED** [review]

**Review by petersteneteg — APPROVED** [review]

File: modules/brushingandlinking/src/brushingandlinkingmanager.cpp (hunk 2 above)
Comment (martinfalk) [reply modules/brushingandlinking/src/brushingandlinkingmanager.cpp:79]: that is handled internally by `filtered_`

File: modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h (hunk 1 above)
Comment (martinfalk) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: This was based on the fact that none of the set functions in the manager actually consider the source. This is only of importance for the events, not the manager.
https://github.com/inviwo/inviwo/blob/7ba4f7571fc2d0a3a2957cc2f6ad3ba23e6684e1/modules/brushingandlinking/src/brushingandlinkingmanager.cpp#L63-L75

re "public", that was the idea.

Comment (petersteneteg) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: the filter does consider source...

Comment (martinfalk) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: However, the source is only used as key in the unordered map which is going to be cleared.

Comment (petersteneteg) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: But will all the sources of filtering get updated when cleared like this

Comment (martinfalk) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: I assume so. Maybe @r-englund can shed some light on this.

Comment (petersteneteg) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: The filtering comes from the pcp plot for example, from each axis range. How is the pcp plot supposed to know that is should reset all the axis ranges when this happens?!? I think this is quite broken now

Comment (r-englund) [reply modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h:68]: What are the use of clear? Who/what is calling it?

**Pull request merged**